internal/storage/storage_handle_test.go

```diff
@@ -98,28 +86,69 @@ func (t *StorageHandleTest) TestBucketHandleWhenBucketDoesNotExistWithNonEmptyBi
 }
 
 func (t *StorageHandleTest) TestNewStorageHandleHttp2Disabled() {
-	sc := getDefaultStorageClientConfig() // by default http1 enabled
+	sc := storageutil.GetDefaultStorageClientConfig() // by default http1 enabled
 
 	t.invokeAndVerifyStorageHandle(sc)
 }
 
 func (t *StorageHandleTest) TestNewStorageHandleHttp2Enabled() {
-	sc := getDefaultStorageClientConfig()
+	sc := storageutil.GetDefaultStorageClientConfig()
 	sc.ClientProtocol = mountpkg.HTTP2
 
 	t.invokeAndVerifyStorageHandle(sc)
 }
 
 func (t *StorageHandleTest) TestNewStorageHandleWithZeroMaxConnsPerHost() {
-	sc := getDefaultStorageClientConfig()
+	sc := storageutil.GetDefaultStorageClientConfig()
 	sc.MaxConnsPerHost = 0
 
 	t.invokeAndVerifyStorageHandle(sc)
 }
 
 func (t *StorageHandleTest) TestNewStorageHandleWhenUserAgentIsSet() {
-	sc := getDefaultStorageClientConfig()
+	sc := storageutil.GetDefaultStorageClientConfig()
 	sc.UserAgent = "gcsfuse/unknown (Go version go1.20-pre3 cl/474093167 +a813be86df) appName (GPN:Gcsfuse-DLC)"
 
 	t.invokeAndVerifyStorageHandle(sc)
 }
+func (t *StorageHandleTest) TestNewStorageHandleWithCustomEndpoint() {
+	url, err := url.Parse(storageutil.CustomEndpoint)
+	AssertEq(nil, err)
+	sc := storageutil.GetDefaultStorageClientConfig()
+	sc.CustomEndpoint = url
+
+	t.invokeAndVerifyStorageHandle(sc)
+}
+
+// This will fail while fetching the token-source, since key-file doesn't exist.
+func (t *StorageHandleTest) TestNewStorageHandleWhenCustomEndpointIsNil() {
+	sc := storageutil.GetDefaultStorageClientConfig()
+	sc.CustomEndpoint = nil
+
+	handleCreated, err := NewStorageHandle(context.Background(), sc)
+
+	AssertNe(nil, err)
+	ExpectThat(err, oglematchers.Error(oglematchers.HasSubstr("no such file or directory")))
+	AssertEq(nil, handleCreated)
+}
+
+func (t *StorageHandleTest) TestNewStorageHandleWhenKeyFileIsEmpty() {
+	sc := storageutil.GetDefaultStorageClientConfig()
+	sc.KeyFile = ""
+
+	t.invokeAndVerifyStorageHandle(sc)
+}
+
+func (t *StorageHandleTest) TestNewStorageHandleWhenReuseTokenUrlFalse() {
+	sc := storageutil.GetDefaultStorageClientConfig()
+	sc.ReuseTokenFromUrl = false
+
+	t.invokeAndVerifyStorageHandle(sc)
+}
+
+func (t *StorageHandleTest) TestNewStorageHandleWhenTokenUrlIsSet() {
```

Comment:
We can add one test to show that NewStorageHandle fails when client protocol is neither HTTP1 or HTTP2 until we support GRPC ?

Reply:
As discussed offline, we can skip this. Right now, if user pass other than HTTP1 or HTTP2 it creates default client doesn't give any error.